src/models/connectionCredentials.ts

```diff
@@ -118,7 +118,8 @@ export class ConnectionCredentials implements IConnectionCredentials {
                     // If this is a profile, and the user has set save password to true and stored the password in the config file,
                     // then transfer the password to the credential store
                     if (profile.savePassword && !wasPasswordEmptyInConfigFile) {
-                        connectionStore.removeProfile(profile).then(() => {
+                        // Remove profile, but keep credential store if savePassword was enabled
+                        connectionStore.removeProfile(profile, profile.savePassword).then(() => {
```

Comment:
Minor: profile.SavePassword will always be true in this scenario, so you might as well always pass true.